applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c

```diff
@@ -273,51 +302,130 @@ int le_audio_config_get(uint32_t *bitrate, uint32_t *sampling_rate)
 
 int le_audio_volume_up(void)
 {
+	int ret;
+
+	if (!playing_audio) {
+		return -ECANCELED;
+	}
+
+	ret = hw_codec_volume_increase();
+	if (ret) {
+		return ret;
+	}
+
 	return 0;
 }
 
 int le_audio_volume_down(void)
 {
+	int ret;
+
+	if (!playing_audio) {
+		return -ECANCELED;
+	}
+
+	ret = hw_codec_volume_decrease();
+	if (ret) {
+		return ret;
+	}
+
 	return 0;
 }
 
 int le_audio_volume_mute(void)
 {
+	int ret;
+
+	if (!playing_audio) {
+		return -ECANCELED;
+	}
+
+	ret = hw_codec_volume_mute();
+	if (ret) {
+		return ret;
+	}
+
 	return 0;
 }
 
 int le_audio_play(void)
 {
+	int ret;
+
+	if ((broadcast_sink == NULL) || playing_audio) {
+		return -ECANCELED;
+	}
+
+	ret = bt_audio_broadcast_sink_sync(broadcast_sink, bis_index_bitfield, streams,
+					   &lc3_preset.codec, NULL);
+	if (ret) {
+		return ret;
+	}
+
 	return 0;
 }
 
 int le_audio_pause(void)
 {
+	int ret;
+
+	if ((broadcast_sink == NULL) || !playing_audio) {
+		return -ECANCELED;
+	}
+
+	ret = bt_audio_broadcast_sink_stop(broadcast_sink);
+	if (ret) {
```

Comment:
Just return ret directly, unless you want to add a LOG_ERR in the if

Reply:
Oh, this is a classic. Improved this now